### ISSUE TITLE
What caused error "some models (but not all) have scaled test statistics"?

### ISSUE THREAD

**Issue by wanglu2014** [yrosseel/lavaan#161]
Thank for your attention. The code was pasted below.
# install.packages('lavaan') 
# install.packages('ggplot2')
# install.packages('reshape2')

# install.packages('lavaan')  #grab package from the source
# install.packages('qpcR')    #grab package from the source
library(lavaan)               #load package
library(qpcR)                 #load package
library(ggplot2)              #load package

#Model comparison: Here we simulate data under one model (A) and fit two
#competing models (A and B). We then use model comparison to test which one
#best fits the data

#True model
samplesize<-500
ULCS_simulate<-'

#####     and should not generally be modified

COG_T2 ~ 1*COG_T1     # Fixed regression of COG_T2 on COG_T1
dCOG1 =~ 1*COG_T2     # Fixed regression of dCOG1 on COG_T2
COG_T2 ~ 0*1          # This line constrains the intercept of COG_T2 to 0
COG_T2 ~~ 0*COG_T2    # This fixes the variance of the COG_T2 to 0  

###### Values can be modified manually to examine the effect on the model

dCOG1 ~ 40*1        # This fixes the intercept of the change score to 10 
COG_T1 ~ 50*1       # This fixes the intercept of COG_T1 to 50 

dCOG1 ~~ 5*dCOG1    # This fixes the variance of the change scores to 5. 
COG_T1 ~~ 8*COG_T1  # This fixes the variance of the COG_T1 to 8. 

dCOG1~-0.1*COG_T1   # This fixes the self-feedback parameter to -0.1. 
'
simdatULCS<-simulateData(ULCS_simulate,sample.nobs = samplesize,meanstructure = T) #Simulate data

#We first fit the true model
ULCS_A<-'

COG_T2 ~ 1*COG_T1     # Fixed regression of COG_T2 on COG_T1
dCOG1 =~ 1*COG_T2     # Fixed regression of dCOG1 on COG_T2
COG_T2 ~ 0*1          # This line constrains the intercept of COG_T2 to 0
COG_T2 ~~ 0*COG_T2    # This fixes the variance of the COG_T2 to 0 

dCOG1 ~ 1             # This estimates the intercept of the change scores 
COG_T1 ~  1           # This estimates the intercept of COG_T1 
dCOG1 ~~  dCOG1       # This estimates the variance of the change scores 
COG_T1 ~~   COG_T1    # This estimates the variance of COG_T1 
dCOG1~COG_T1          # This estimates the self-feedback parameter

'

fitULCS_A <- lavaan(ULCS_A, data=simdatULCS, estimator='mlr',fixed.x=FALSE,missing='fiml')
summary(fitULCS_A, fit.measures=TRUE, standardized=TRUE, rsquare=TRUE)

#We now fit a competing, simpler model that constraints the autoregressive
#parameter to 0

ULCS_B<-'

COG_T2 ~ 1*COG_T1     # Fixed regression of COG_T2 on COG_T1
dCOG1 =~ 1*COG_T2     # Fixed regression of dCOG1 on COG_T2
COG_T2 ~ 0*1          # This line constrains the intercept of COG_T2 to 0
COG_T2 ~~ 0*COG_T2    # This fixes the variance of the COG_T2 to 0 

dCOG1 ~ 1             # This estimates the intercept of the change scores 
COG_T1 ~  1           # This estimates the intercept of COG_T1 
dCOG1 ~~  dCOG1       # This estimates the variance of the change scores 
COG_T1 ~~   COG_T1    # This estimates the variance of COG_T1 
dCOG1~0*COG_T1          # This estimates the self-feedback parameter

'

fitULCS_B <- lavaan(ULCS_B, data=simdatULCS, estimator='mlr',fixed.x=FALSE,missing='fiml')
summary(fitULCS_B, fit.measures=TRUE, standardized=TRUE, rsquare=TRUE)

#Likelihood ratio test:
anova(fitULCS_A,fitULCS_B)

**Comment by TDJorgensen** [comment]
Hello @wanglu2014, 
You might notice your syntax does not appear as you intended.  Asterisks `*` and hash marks `#` disappeared because they are special characters used to format the text in your message (`*` is used for italics, `#` is used to make "section headers").  If you click on the <> button at the top of the text editor, it will insert "backticks" for you to embed in-line syntax.  For multi-line code, you should type 3 backticks on one line
```
Like this: ```
```
click Enter/Return, then 3 backticks on another line.  Paste your code between them, and the whole set of code will be properly formatted.  FYI, we don't need the other packages for your example, only `lavaan`.


@yrosseel, 
I thought this issue was resolved, but apparently not for a saturated model (i.e., *df* = 0). Only in that case, the `$test` is still named "standard" in the lavaan object's `@test` slot
```
samplesize<-500

ULCS_simulate <- '
COG_T2 ~ 1*COG_T1 # Fixed regression of COG_T2 on COG_T1
dCOG1 =~ 1*COG_T2 # Fixed regression of dCOG1 on COG_T2
COG_T2 ~ 0*1 # This line constrains the intercept of COG_T2 to 0
COG_T2 ~~ 0*COG_T2
dCOG1 ~ 40*1 # This fixes the intercept of the change score to 10
COG_T1 ~ 50*1 # This fixes the intercept of COG_T1 to 50
dCOG1 ~~ 5*dCOG1 # This fixes the variance of the change scores to 5.
COG_T1 ~~ 8*COG_T1 # This fixes the variance of the COG_T1 to 8.
dCOG1~-0.1*COG_T1 # This fixes the self-feedback parameter to -0.1.
'
simdatULCS<-simulateData(ULCS_simulate,sample.nobs = samplesize,meanstructure = T)

ULCS_A <- '
COG_T2 ~ 1*COG_T1 # Fixed regression of COG_T2 on COG_T1
dCOG1 =~ 1*COG_T2 # Fixed regression of dCOG1 on COG_T2
COG_T2 ~ 0*1 # This line constrains the intercept of COG_T2 to 0
COG_T2 ~~ 0*COG_T2 # This fixes the variance of the COG_T2 to 0
dCOG1 ~ 1 # This estimates the intercept of the change scores
COG_T1 ~ 1 # This estimates the intercept of COG_T1
dCOG1 ~~ dCOG1 # This estimates the variance of the change scores
COG_T1 ~~ COG_T1 # This estimates the variance of COG_T1
dCOG1~COG_T1 # This estimates the self-feedback parameter
'
fitULCS_A <- lavaan(ULCS_A, data=simdatULCS, estimator='mlr',fixed.x=FALSE,missing='fiml')
fitULCS_A@test
```

@wanglu2014,
This issue only occurs because `fitULCS_A` has no degrees of freedom.  So the LRT between these 2 models is simply the LRT of `fitULCS_B`, which has degrees of freedom.  We will fix the bug, but you don't need to compare these 2 models to test their difference, just test Model B.

```
lavTestLRT(fitULCS_B)
```

**Comment by yrosseel** [comment]
Fixed in lavaan dev 0.6-6.1530
